src/layman/map/rest_maps_authz_test.py

```diff
@@ -0,0 +1,69 @@
+import json
+from test import process, process_client
+liferay_mock = process.liferay_mock
+
+
+def test_map_with_unauthorized_layer(liferay_mock):
+    username1 = 'test_map_with_unauthorized_layer_user1'
+    layername1 = 'test_map_with_unauthorized_layer_layer1'
+    mapname1 = 'test_map_with_unauthorized_layer_map1'
+    username2 = 'test_map_with_unauthorized_layer_user2'
+    layername2 = 'test_map_with_unauthorized_layer_layer2'
+
+    layman_process = process.start_layman(process.AUTHN_SETTINGS)
+
+    user1_authz_headers = process_client.get_authz_headers(username1)
+    user2_authz_headers = process_client.get_authz_headers(username2)
+
+    process_client.reserve_username(username1, headers=user1_authz_headers)
+    process_client.reserve_username(username2, headers=user2_authz_headers)
+
+    process_client.publish_layer(username1, layername1, headers=user1_authz_headers)
+    process_client.publish_layer(username2, layername2, headers=user2_authz_headers)
+
+    # publish map composed of layers of both users, read for everyone
+    process_client.publish_map(
+        username1,
+        mapname1,
+        file_paths=['sample/layman.map/internal_url_unauthorized_layer.json'],
+        access_rights={
+            'read': 'EVERYONE',
+            'write': f"{username1},{username2}",
+        },
+        headers=user1_authz_headers,
+    )
+    process_client.assert_user_maps(username1, [mapname1], headers=user1_authz_headers)
+    process_client.assert_user_maps(username1, [mapname1], headers=user2_authz_headers)
+
+    layer1_uuid = process_client.get_layer(username1, layername1, headers=user1_authz_headers)['uuid']
+    layer2_uuid = process_client.get_layer(username2, layername2, headers=user2_authz_headers)['uuid']
+
+    def assert_operates_on(workspace, mapname, expected_layers, authz_headers):
```

Comment:
Vnitřní metoda by podle mne měla být definovaná na začátku. Když je takhle uprostřed, tak je to pro mne nepřehledné. Byť souhlasím s hodnotou, že to je blízko volání.